src/protocol/boolean/random_bits.rs

```diff
@@ -0,0 +1,129 @@
+use crate::error::BoxError;
+use crate::ff::{Field, Int};
+use crate::protocol::boolean::BitOpStep;
+use crate::protocol::modulus_conversion::convert_shares::{ConvertShares, XorShares};
+use crate::protocol::{context::ProtocolContext, RecordId};
+use crate::secret_sharing::Replicated;
+use futures::future::try_join_all;
+
+/// This protocol generates a sequence of uniformly random sharing of bits in `F_p`.
+/// Adding these 3-way secret-sharing will yield the secret `r_i ∈ {0,1}`.
+pub struct RandomBits {}
+
+impl RandomBits {
+    #[allow(dead_code)]
+    pub async fn execute<F: Field>(
+        ctx: ProtocolContext<'_, Replicated<F>, F>,
+        record_id: RecordId,
+    ) -> Result<(Vec<Replicated<F>>, Replicated<F>), BoxError> {
+        // We assume the bit length we operate in would not exceed 255
+        #[allow(clippy::cast_possible_truncation)]
+        let l = F::Integer::BITS as u8;
+
+        // Generate a replicated random field values. We'll use these fields as
+        // the source of `l`-bit long uniformly random sequence of bits.
+        let r_p: Replicated<F> = ctx.prss().generate_replicated(record_id);
```

Comment:
Since you're just going to immediately call as_u128() I don't see any reason to coerce these into field values. Just use the raw numbers you get from generate_values()

Reply:
I used `generated_replicated()` because the generated random numbers must be less than `p`. I changed it to `generate_fields` since we don't really need `Replicated`.

Reply:
Changed to use `generate_values()`